Altis_Life.Altis/core/cop/fn_copMarkers.sqf

```diff
@@ -3,44 +3,34 @@
     Author: Bryan "Tonic" Boardwine
 
     Description:
-    Marks cops on the map for other cops. Only initializes when the actual map is open.
+    Marks cops on the map for other cops.
 */
-
 private _markers = [];
-private _cops = [];
-
-sleep 0.5;
-if (visibleMap) then {
-    _cops = playableUnits select {side _x isEqualTo west};
-
-    //Create markers
-    {
-        if !(_x isEqualTo player) then {
-            private _marker = createMarkerLocal [format ["%1_marker",_x],visiblePosition _x];
-            _marker setMarkerColorLocal "ColorBLUFOR";
-            _marker setMarkerTypeLocal "Mil_dot";
-            _marker setMarkerTextLocal format ["%1", _x getVariable ["realname",name _x]];
 
-            _markers pushBack [_marker,_x];
-        };
-        true
-    } count _cops;
-
-    while {visibleMap} do {
-        {
-            _x params ["_marker","_unit"];
-            if (!isNil "_unit" && {!isNull _unit}) then {
-                _marker setMarkerPosLocal (visiblePosition _unit);
-            };
-            true
-        } count _markers;
-        if (!visibleMap) exitWith {};
-        sleep 0.02;
+{
+    if (side _x isEqualTo west) then {
+        private _marker = createMarkerLocal [format ["%1_marker",_x],getPosATL _x];
+        _marker setMarkerColorLocal "ColorBLUFOR";
+        _marker setMarkerTypeLocal "Mil_dot";
+        _marker setMarkerTextLocal format ["%1", _x getVariable ["realname",name _x]];
+        _markers pushBack [_marker,_x];
     };
+} forEach (playableUnits - [player]);
 
+while {visibleMap} do {
     {
-        deleteMarkerLocal (_x select 0);
-    } count _markers;
-    _markers = [];
-    _cops = [];
+        _x params [
+            ["_mark",""],
+            ["_unit",objNull]
+        ];
+        if !(isNull _unit) then {
```

Comment:
but what if it is? what if someone dies/respawns? (same for previous file)

Reply:
Just avoids moving the marker to [0,0,0] until they've closed the map it seems.